wolfcrypt/src/evp.c

```diff
@@ -4124,9 +4139,10 @@ int wolfSSL_EVP_MD_type(const WOLFSSL_EVP_MD *md)
             }
             ctx->gcmAuthInSz = 0;
 #endif
+            return WOLFSSL_SUCCESS;
         }
 
-        return WOLFSSL_SUCCESS;
+        return WOLFSSL_FAILURE;
```

Comment:
OpenSSL man page for EVP_CIPHER_CTX_cleanup() states that it returns 1 on success.  Looking at the OpenSSL 1.1.1j source code EVP_CIPHER_CTX_cleanup() maps to EVP_CIPHER_CTX_reset(), which returns 1 if the input "ctx" is NULL.  This means we would still return WOLFSSL_SUCCESS (1) here.

If this is different than expected, maybe it varies with version of OpenSSL?

Reply:
openSSL 1.0.2 returns "1" also. I revert this change.